server/channels/api4/status.go

```diff
@@ -105,8 +107,14 @@ func updateUserStatus(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 
 	currentStatus, err := c.App.GetStatus(c.Params.UserId)
-	if err == nil && currentStatus.Status == model.StatusOutOfOffice && status.Status != model.StatusOutOfOffice {
-		c.App.DisableAutoResponder(c.AppContext, c.Params.UserId, c.IsSystemAdmin())
+	if err != nil {
+		c.Logger.Warn("Failed to get current status", mlog.Err(err))
+	} else if currentStatus.Status == model.StatusOutOfOffice && status.Status != model.StatusOutOfOffice {
+		err = c.App.DisableAutoResponder(c.AppContext, c.Params.UserId, c.IsSystemAdmin())
+		if err != nil {
+			c.Err = err
+			return
+		}
```

Comment:
I think we should log here again. Returning from here would change behavior from before.

Reply:
@agnivade done!